envpool/core/array.h

```diff
@@ -198,4 +220,10 @@ class Array {
   [[nodiscard]] std::shared_ptr<char> SharedPtr() const { return ptr_; }
 };
 
+template <typename Dtype>
+class TArray : public Array {
```

Comment:
Do we really need TArray at this point?

Reply:
You're right, it can still work without TArray.